libp2p/NodeTable.h

```diff
@@ -404,18 +410,21 @@ struct DiscoveryDatagram: public RLPXDatagramFace
     NodeID sourceid; // sender public key (from signature)
     h256 echo;       // hash of encoded packet, for reply tracking
 
-    // All discovery packets carry a timestamp, which must be greater
+    // Most discovery packets carry a timestamp, which must be greater
     // than the current local time. This prevents replay attacks.
-    uint32_t ts = 0;
-    bool isExpired() const { return secondsSinceEpoch() > ts; }
+    boost::optional<uint32_t> timestamp;
```

Comment:
Because now not all of the packets contain timestamp - `ENRResponse` doesn't.
It won't open us up to replay attacks, because we still parse it and check it for all packets that should have it.

Reply:
Ah you're right, plus there's no danger of replay attacks with the response since ENRs have the sequence number and the hash of the request.